tests/deployments/test_cli.py

```diff
@@ -84,14 +84,13 @@ def test_get():
 )
 def test_predict(tmpdir):
     temp_input_file_path = tmpdir.join("input.json").strpath
-    with open(temp_input_file_path, "w+t") as temp_input_file:
+    with open(temp_input_file_path, "w") as temp_input_file:
         temp_input_file.write('{"data": [5000]}')
-        temp_input_file.seek(0)
-        runner = CliRunner()
-        res = runner.invoke(
-            cli.predict, ["--target", f_target, "--name", f_name, "--input-path", temp_input_file]
-        )
-        assert "1" in res.stdout
+    runner = CliRunner()
+    res = runner.invoke(
+        cli.predict, ["--target", f_target, "--name", f_name, "--input-path", temp_input_file_path],
+    )
+    assert "1" in res.stdout
```

Comment:
This test failed with:

```
    @pytest.mark.skipif(
        "MLFLOW_SKINNY" in os.environ,
        reason="Skinny Client does not support predict due to the pandas dependency",
    )
    def test_predict(tmpdir):
        temp_input_file_path = tmpdir.join("input.json").strpath
        with open(temp_input_file_path, "w") as temp_input_file:
            temp_input_file.write('{"data": [5000]}')
            runner = CliRunner()
            res = runner.invoke(
                cli.predict,
                ["--target", f_target, "--name", f_name, "--input-path", temp_input_file_path],
            )
>           assert "1" in res.stdout
E           AssertionError: assert '1' in ''
E            +  where '' = <Result ValueError('Expected object or value')>.stdout
```

Reply:
I haven't figured out why this test started failing with the latest version of `click` (which might have changed how to process arguments for `CliRunner.invoke`)

Reply:
I think passing `temp_input_file_path` makes more sense rather than passing `temp_input_file` which is a file handler.